shm/inc/Cargo.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <string>
+class Cargo {
+public:
+    Cargo(std::string, size_t, size_t);
+    std::string getName() const { return name_; }
+    size_t getAmount() const { return amount_; }
+    size_t getBestPrice() const { return basePrice_; }
+
+private:
+    const std::string name_ { }; 
```

Comment:
No need for curly braces. Default std::string initialization is "" and size of zero.